src/ucp/wireup/select.c

```diff
@@ -806,6 +807,32 @@ ucp_wireup_ep_init_flags(const ucp_wireup_select_params_t *select_params,
     return select_params->ep_init_flags | select_ctx->ucp_ep_init_flags;
 }
 
+static ucs_status_t
+ucp_wireup_add_cm_lane(const ucp_wireup_select_params_t *select_params,
+                       ucp_wireup_select_context_t *select_ctx)
+{
+    ucp_wireup_select_info_t select_info;
+
+    if (!(select_params->ep_init_flags & (UCP_EP_INIT_CM_WIREUP_CLIENT |
+                                          UCP_EP_INIT_CM_WIREUP_SERVER))) {
+        return UCS_OK;
+    }
+
+    select_info.reachable  = 1;  /**< UCT CM provides reachability by its API
+                                      CBs */
+    select_info.priority   = 0;  /**< Currently we have only 1 CM
+                                      implementation */
+    select_info.rsc_index  = UCP_NULL_RESOURCE; /**< RSC doesn't matter for CM */
+    select_info.addr_index = 0;  /**< This makes sense only for transport
+                                      lanes */
+    select_info.score      = 0.; /**< TODO: when we have > 1 CM implementation */
+
+    /* server is not a proxy because it can create all lanes connected */
```

Comment:
what does this comment mean in this context?

Reply:
it describes meaning of 4th arg `is_proxy`:
```
static UCS_F_NOINLINE void
ucp_wireup_add_lane_desc(const ucp_wireup_select_info_t *select_info,
                         ucp_rsc_index_t dst_md_index,
                         uint32_t usage, int is_proxy,
                         ucp_wireup_select_context_t *select_ctx)
```

Reply:
why are client and server different in this matter?

Reply:
because client TL lanes can't be connected until server response due to lack of remote address. Need to have proxy lane to keep pending queue.